gridt/src/app/core/settings.service.ts

```diff
@@ -17,44 +17,46 @@ export interface ServerMessage {
   providedIn: "root"
 })
 export class SettingsService {
-  /*
-   * Subscribe to this observable to ready the API.
-   */
-  public URL = "https://api.gridt.org";
 
   constructor (
     private http: HttpClient,
     private auth: AuthService,
     private secStore: SecureStorageService,
-    // private api: ApiService
+    private api: ApiService
   ) {}
 
+  private emptySettings: Settings = {
+    identity: {
+      id: 0,
+      username: "",
+      bio: "",
+      avatar: "",
+      email: ""
+    }
+  }
+
   /**
    * Create ReplaySubject which yields the latest user settings.
+   * The depth of the ReplaySubject is set to 1, which makes sure only the
+   * latest available settings are retured to the user.
    */
-  public _user_settings$ = new ReplaySubject<Settings>(1);
-  // The depth of the ReplaySubject is set to 1, which makes sure only the
-  // latest available settings are retured to the user.
+  private _user_settings$ = new ReplaySubject<Settings>(1);
 
-  /**
-   * Create a pip which skips the trivial updates of _user_settings$
-   */
-  public special_pipe$: UnaryFunction<Observable<Settings>, Observable<Settings>> =
-    pipe(
-      skip(1),                // Skip default
-      distinctUntilChanged(), // Register the server response
-      // filter(val => !!val ),  // Skip no changes in the server setting
-    );
+  public set_user_settings(settings): void {
+    this._user_settings$.next(settings);
+  };
 
   /**
    * Transforms the user settings Subject to an Observable
    */
   get the_user_settings$ (): Observable<Settings> {
     // If the user_settings change, make sure the new settings are stored
     // in the localstorage.
+    let cachedValue: Settings;
     if (this.auth.isLoggedIn$.subscribe()) {

```

Comment:
This if statement will always pass, because the `Observable.subscribe` always returns a subscription. I could suggest piping the `isLoggedIn$` observable. Also right now you are not making use of the cachedValue variable.

Reply:
By the fact that all tests pass, I can tell that this is not a feature that is tested at the moment. I would recommend writing the test first and then implementing this feature.